lib/modules/showImages.js

```diff
@@ -817,8 +817,14 @@ addModule('showImages', (module, moduleID) => {
 		updateParentHeight(expandoButton);
 	}
 
-	function generateImageExpando(expandoButton) {
+	async function generateImageExpando(expandoButton) {
 		const imageLink = expandoButton.imageLink;
+
+		// Handle init if needed
+		if (typeof imageLink.onFirstExpand === 'function') {
```

Comment:
Re [my comment about preloading metadata](https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/2818#issuecomment-201909688), I'd vote for a function name more like "fetchLinkMetadata".